docs/checkit.js

```diff
@@ -777,55 +786,118 @@ CheckIt.prototype.validateInput = function ($form, $inputFormGroup, inputId) {
         
         validateEndDate = this.validateInput(this.$fullForm, this.$endDateFormGroup, 'endDate'),
     
         isValid = validateTitle && validateStartDate && validateEndDate;
-    
+
     // Make sure input is OK before parsing and validating dates
     return (isValid && this.validateDates(startDateString, endDateString, this.$endDateFormGroup));
    
 };
 
-CheckIt.prototype.createCalendar = function () {
-    // Create a new calendar and display it.
+CheckIt.prototype.createCalendar = function (calState) {
+    // creates a calendar
+    
     "use strict";
-    var title = this.$calendarTitle.val(),
-        start = this.$startDate.val(),
-        end = this.$endDate.val(),
-        state,
-        calendar,
+    var calendar,
         initP,
         buildP;
-        
+    //if (this.validateForm(start, end)) {
     
-    if (this.validateForm(start, end)) {
-        
-        //clear the previously displayed calendar
-        this.clearCalendarDiv();
+    //clear the previously displayed calendar
+    this.clearCalendarDiv();
 
-        //make a calendar State
-        state = emptyCalendarState({startDate: start, endDate: end, calendarTitle: title});
-        
-        //make calendar object
+    //make calendar object
 
-        calendar = new Calendar(state);
-        
-        // Initialize calendar in the storage
-        initP = this.store.initializeCalendar(calendar);
+    calendar = new Calendar(calState);
+
+    // Initialize calendar in the storage
+    initP = this.store.initializeCalendar(calendar);
+
+    //add calendar to dropdown
+    this.addCalendarToDropdown(calendar.state.uniqueId, calendar.state.title,
+        this.$calendarDropdown);
 
-        //add calendar to dropdown
-        this.addCalendarToDropdown(calendar.state.uniqueId, calendar.state.title,
-            this.$calendarDropdown);
+    //build the calendar
+    buildP = this.buildCalendar(calendar);
+    this.hideForm(this.$buildCalendarForm);
     
-        //build the calendar
-        buildP = this.buildCalendar(calendar);
-        this.hideForm(this.$buildCalendarForm);
+    return Promise.all([initP, buildP]);
+    //}
+};
+
+CheckIt.prototype.editCalendar = function (title, start, end, activeCalId) {
+    // Edits the calendar currently on display (current active calendar)
+    // and displays the edited version
+    "use strict";
+    
+    var checkit = this,
+        startDate = moment(start, "YYYY-MM-DD"),
+        endDate = moment(end, "YYYY-MM-DD"),
+        loadP;
+    
+    // I want to return a non-pending promise
+    
+    return checkit.store.loadById(activeCalId)
+        .then(function (calState) {
+            calState.endDateString = endDate.format("YYYYMMDD");
+            calState.startDateString = startDate.format("YYYYMMDD");
+            calState.title = title;
+            checkit.createCalendar(calState);
+        });
+};
+
+
+CheckIt.prototype.editOrCreate = function (params) {
```

Comment:
I think this function does too much work and is hard to understand.

Seems like what you want is a boolean function that tells you whether or not there is an active calendar (which would always imply that it's displayed) and then based on that create one or not. Then edit it if needed.

Reply:
Yeah but it works. So. Who cares? I thought that's how things were done. 😆

Reply:
You can definitely just leave it if it works. It will just be harder to understand the next time you come look at it to fix a bug or update functionality.